### PR TITLE
EWS O365 version using MSAL lib for ROPC flow

### PR DESCRIPTION
**Pull request by hiep4hiep**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [X] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation 

## Docker image required
/docker_image_create name=demisto-ews-msal:2.1 base="demisto/py3ews:1.0.0.45234" dependencies=msal,urllib3==1.26.8

### CODE REVIEW

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 1]
```diff
@@ -4114,6 +4190,157 @@ def get_cve_command(args: dict) -> list[CommandResults]:
...
+        :param after:   On your first request, don't provide an after token. On subsequent requests, 
+                        provide the after token from the previous response to continue from that place in the results.
+        :param limit: The maximum number of result to return.
+        :param time_range: The time when the change was reported to the CrowdStrike cloud. Define with FQL.
+    """
+    after = args.get('after')
+    limit = args.get('limit', 1000)
+    time_range = args.get('time_range', 'now-1m')
+    try:
+        raw_res = get_spotlight_vulnerabilities_list(after=None, limit=1000, time_range='now-1m')
```
Comment (adi88d) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:4281]:
```suggestion
        raw_res = get_spotlight_vulnerabilities_list(after= after, limit=limit, time_range= time_range)
```

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.yml [hunk 2]
```diff
@@ -2896,6 +2896,107 @@ script:
...
+    - contextPath: CrowdStrike.Spotlight.VulnerabilityDetails.created_timestamp
+      description: Timestamp when the vulnerability was detected.
+      type: String
+    - contextPath: CrowdStrike.Spotlight.VulnerabilityDetails.cve
+      description: CVE Number.
+      type: String
+    - contextPath: CrowdStrike.Spotlight.VulnerabilityDetails.remediation
+      description: The remediation method.
+      type: String
+
```
Comment (adi88d) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.yml:2999]: remove the empty line
```suggestion
```

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 3]
```diff
@@ -4114,6 +4190,157 @@ def get_cve_command(args: dict) -> list[CommandResults]:
...
+        return str(e)
+
+
+def get_spotlight_vulnerabilities_details_command(args: dict):
+    """
+        Get the full list of properties about one or more changes using the ID of each change
+
+        :param vul_id: The vulnerability ID to get details
+    """
+    vul_id = str(args.get('vul_id'))
```
Comment (adi88d) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:4311]: there is no argument with the name `vul_id`
```suggestion
    vul_id = str(args.get('change_id'))
```

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon_test.py [hunk 4]
```diff
@@ -4291,3 +4291,325 @@ def test_cs_falcon_spotlight_search_vulnerability_host_by_command(mocker):
...
+                    "spotlight_published_date":"2022-08-09T20:00:00Z",
+                    "actors":[
+                    "GOBLIN PANDA",
+                    "STARDUST CHOLLIMA"
+                    ],
+                    "name":"Dejablue"
+                },
+                "host_info":{
+                    "hostname":"CSTEST_Spotlight_hostname",
+                    "local_ip":"192.000.0.2",
```
Comment (adi88d) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon_test.py:4554]:
```suggestion
                    "local_ip":"127.0.0.1",
```

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon_test.py [hunk 5]
```diff
@@ -4291,3 +4291,325 @@ def test_cs_falcon_spotlight_search_vulnerability_host_by_command(mocker):
...
+                "cve":{
+                    "id":"CVE-2013-0000",
+                    "base_score":7.5,
+                    "severity":"HIGH",
+                    "exploit_status":60,
+                    "exprt_rating":"HIGH",
+                    "description":"The parseRTSPRequestString function in Live Networks Live555 Streaming Media fix for CVE-2013-XXXX.",
+                    "published_date":"2014-01-23T21:55:00Z",
+                    "vendor_advisory":[
+                    "http://www.live.com/liveMedia/public/changelog.txt"
```
Comment (adi88d) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon_test.py:4531]:
```suggestion
                    "http://www.test.com/liveMedia/public/changelog.txt"
```

**Review by adi88d — CHANGES_REQUESTED** [review]
Hi @hiep4hiep ,Thank you for your contribution!

- Please update the **CrowdStrikeFalcon/README.md** file with the commands you added, 
To do that, please run the `demisto-sdk generate-docs` command (use the -e flag for using a commad_examples file) - for more information please see: https://xsoar.pan.dev/docs/concepts/demisto-sdk#generate-docs.
- see my other comments below

File: Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py [hunk 6]
```diff
@@ -0,0 +1,2625 @@
...
+from oauthlib.oauth2 import OAuth2Token
+from exchangelib.version import EXCHANGE_O365
+from exchangelib.protocol import BaseProtocol, NoVerifyHTTPAdapter
+
+#from msal import authority
+#from Base.Scripts.CommonServerPython.CommonServerPython import CommandResults, get_integration_context, handle_proxy, return_results, tableToMarkdown
+#from Base.Scripts.CommonServerPython.CommonServerPython import set_integration_context
+#
+#
+#
```
Comment (adi88d) [inline Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py:64]: remove
```suggestion

```

File: Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py [hunk 7]
```diff
@@ -0,0 +1,2625 @@
...
+
+import requests
+import traceback
+from typing import Dict, Any
+import sys
+import json
+import logging
+import msal
+import os
```
Comment (adi88d) [inline Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py:73]: remove unused code
```suggestion

```

File: Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py [hunk 8]
```diff
@@ -0,0 +1,2625 @@
...
+        except Exception as ex:
+            demisto.error("Failed starting Process: {}".format(ex))
+    else:
+        sub_main()
+
+def log_memory():
+    if is_debug_mode():
+        demisto.debug(f'memstat\n{str(subprocess.check_output(["ps", "-opid,comm,rss,vsz"]))}')
+
+from MicrosoftApiModule import *  # noqa: E402
```
Comment (adi88d) [inline Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py:2622]: move to the top
```suggestion
from MicrosoftApiModule import *  # noqa: E402
```

File: Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py [hunk 9]
```diff
@@ -0,0 +1,2625 @@
...
+            p.start()
+            p.join()
+        except Exception as ex:
+            demisto.error("Failed starting Process: {}".format(ex))
+    else:
+        sub_main()
+
+def log_memory():
+    if is_debug_mode():
+        demisto.debug(f'memstat\n{str(subprocess.check_output(["ps", "-opid,comm,rss,vsz"]))}')
```
Comment (adi88d) [inline Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py:2620]: move this method

**Review by adi88d — CHANGES_REQUESTED** [review]
For the EWS_MSAL integration:
- please use CommandResults in the command methods.
- add `example.txt` file with examples to all the commands
- add test playbook

**Comment by hiep4hiep** [comment]
Hi @adi88d , apologize for the mixed up as I am not intended to push the CrowStrikeFalcon pack to this PR because it is the other work in progress.
The content I wanted push is only EWS_MSAL. Please help review that one instead.
Thank you.

File: Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py [hunk 10]
```diff
@@ -0,0 +1,2625 @@
...
+
+class GetSearchableMailboxes(EWSService):
+    """
+    EWSAccountService class used for getting Searchable Mailboxes
+    """
+    SERVICE_NAME = "GetSearchableMailboxes"
+    element_container_name = f"{{{MNS}}}SearchableMailboxes"
+
+    @staticmethod
+    def parse_element(element):
```
[the remark this answers is not in the document]
Comment (hiep4hiep) [reply Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py:454]: Hi @adi88d , this method is from the OOTB version. I did not modify anything.

File: Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py [hunk 11]
```diff
@@ -0,0 +1,2625 @@
...
+            "attachmentContentLocation": attachment.content_location,
+            "attachmentSize": attachment.size,
+            "attachmentLastModifiedTime": attachment.last_modified_time.ewsformat(),
+            "attachmentIsInline": attachment.is_inline,
+            ATTACHMENT_TYPE: FILE_ATTACHMENT_TYPE
+            if isinstance(attachment, FileAttachment)
+            else ITEM_ATTACHMENT_TYPE,
+        }
+    except TypeError as e:
+        if str(e) != "must be string or buffer, not None":
```
[the remark this answers is not in the document]
Comment (hiep4hiep) [reply Packs/EWS/Integrations/EWS_MSAL/EWS_MSAL.py:940]: Hi @adi88d , this method is from the OOTB version. I did not modify anything.

**Comment by adi88d** [comment]
@hiep4hiep we decided to accept your contribution as community support, please change the pull request:
- create new pack with community support level for the **EWS MSAL** integration and revert all the changes in **CrowdStrikeFalcon** and **EWS** packs

**Comment by hiep4hiep** [comment]
Hi @adi88d , I have deleted the CrowdStrikeFalcon and added EWS-MSAL as a community pack. Please help review. Thank you.

File: Packs/EWS-MSAL/Integrations/EWS_MSAL/EWS_MSAL.py [hunk 12]
```diff
@@ -0,0 +1,2625 @@
...
+        try:
+            p = Process(target=process_main)
+            p.start()
+            p.join()
+        except Exception as ex:
+            demisto.error("Failed starting Process: {}".format(ex))
+    else:
+        sub_main()
+
+def log_memory():
```
Comment (adi88d) [inline Packs/EWS-MSAL/Integrations/EWS_MSAL/EWS_MSAL.py:2618]: move this method before the main

File: Packs/EWS-MSAL/Integrations/EWS_MSAL/EWS_MSAL.py [hunk 13]
```diff
@@ -0,0 +1,2625 @@
...
+    if is_debug_mode():
+        demisto.debug(f'memstat\n{str(subprocess.check_output(["ps", "-opid,comm,rss,vsz"]))}')
+
+from MicrosoftApiModule import *  # noqa: E402
```
Comment (adi88d) [inline Packs/EWS-MSAL/Integrations/EWS_MSAL/EWS_MSAL.py:2622]: move to the top

File: Packs/EWS-MSAL/Integrations/EWS_MSAL/EWS_MSAL.py [hunk 14]
```diff
@@ -0,0 +1,2625 @@
...
+from exchangelib.version import EXCHANGE_O365
+from exchangelib.protocol import BaseProtocol, NoVerifyHTTPAdapter
+
+#from msal import authority
+#from Base.Scripts.CommonServerPython.CommonServerPython import CommandResults, get_integration_context, handle_proxy, return_results, tableToMarkdown
+#from Base.Scripts.CommonServerPython.CommonServerPython import set_integration_context
+#
+#
+#
+
```
Comment (adi88d) [inline Packs/EWS-MSAL/Integrations/EWS_MSAL/EWS_MSAL.py:65]: remove

**Review by adi88d — CHANGES_REQUESTED** [review]
Hi @hiep4hiep
- please revert the changes in **CrowStrikeFalcon** pack - you deleted it instead
- delete this image `Packs/EWS-MSAL/Author_image.png`
- solve my other comments

**Comment by adi88d** [comment]
Hi @hiep4hiep,
Thank you for your contribution.
I have not received a response from you in the last two weeks, so I will close your PR now.
Please feel free to re-open it when you are available to continue.
Thanks again.

**Pull request closed without merging**